api/schema/aws_tag_to_label_resource.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Illumio, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package schema
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	awsTagToLabelResource = Resource{
+		TypeName: "aws_tag_to_label",
+		Schema: resource_schema.Schema{
+			Version:     1,
+			Description: "Maps AWS account tags to CloudSecure labels.",
+			Attributes: map[string]resource_schema.Attribute{
+				IDFieldName: idAttribute,
+				"key": resource_schema.StringAttribute{
+					MarkdownDescription: "CloudSecure label key.",
+					Required:            true,
+				},
```

Comment:
Is this mutable?
If not, declare it so:
```suggestion
				"key": StringResourceAttributeWithMode{
					StringAttribute: resource_schema.StringAttribute{
						Description:   "Key of the created CloudSecure labels.",
						Required:      true,
						PlanModifiers: []planmodifier.String{
							stringplanmodifier.RequiresReplace(),
						},
					},
					attributeWithMode: attributeWithMode{
						Mode: ImmutableAttributeMode,
					},
				},
```

Reply:
You added the `RequiresReplace` modifier, but you're still missing the `ImmutableAttributeMode` mode.

Reply:
its editable

Reply:
If it was mutable, then adding the `RequiresReplace` modifier wouldn't make sense.

Reply:
OK, I see that you finally removed the `RequiresReplace` modifier. Now it makes sense.